src/main/java/uk/gov/hmcts/auth/provider/service/api/microservice/MicroserviceRepository.java

```diff
@@ -47,7 +48,7 @@ public Microservice findOne(String microservice) {
         return new Microservice(microserviceLowercase, keys.get(microserviceLowercase));
     }
 
-    public Set<String> getNames() {
-        return keys.keySet();
+    public List<String> getNames() {
+        return keys.keySet().stream().sorted().collect(toList());
```

Comment:
Just use `TreeSet` - provided by java specifically for sorting collections

Reply:
`keys` is a map

Reply:
suggestion was to the set as per previous signature. so

```java
public Set<String> getNames() {
    return new TreeSet<>(keys.keySet());
}

// or even more direct, since TreeSet uses TreeMap
return new TreeMap<>(keys).keySet();
```

Reply:
Given we start from an unsorted set, it doesn't matter - we have to sort the set no matter what result type we pick. If we mutated it later (ie. added and removed elements) it could matter, but that's not our case.